jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ManagedBeanDiagnosticsCollector.java

```diff
@@ -248,7 +246,19 @@ else if (!hasParameterizedInjectConstructor)
                                     CONSTRUCTOR_DIAGNOSTIC_CODE);
                             diagnostics.add(diagnostic);
                         }
-
+                    }   
+                }
+                
+                /**
+                 * If a managed bean class is of generic type, it must be annotated with @Dependent
+                 */
+                if (isManagedBean) {
+                	boolean isClassGeneric = type.getTypeParameters().length != 0;
+                    
+                    if (isClassGeneric && managedBeanAnnotations.stream()
```

Comment:
Refactor the second condition in if block `managedBeanAnnotations.stream().anyMatch(annotation -> !annotation.equals("Dependent"))` to be a boolean var for better readability; similar to isClassGeneric.